tests/test_core.py

```diff
@@ -4553,16 +4553,27 @@ def test_fs_mmap(self):
       self.do_run_from_file(src, out)
 
   def test_fs_errorstack(self):
+    post = '''
+def process(filename):
+  src = open(filename, 'r').read()
+  open(filename, 'w').write('"use strict";\\n' + src)
+'''
+
     Settings.FORCE_FILESYSTEM = 1
     self.do_run(r'''
       #include <emscripten.h>
       int main(void) {
         EM_ASM(
```

Comment:
Thanks, this PR looks good, with

 * Please add a comment in the test about testing strict mode is to prevent problems if a user enables it.
 * Please add to the C++ code here some iostream usage, `#include <iostream>` before and `std::cout << "hello world\n";` before the `EM_ASM` - that way we get strict mode coverage for a nontrivial amount of code, which could help catch future bugs.